testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -56,32 +56,32 @@
   text = Regexp.new(text) if type == 'regex'
   next if has_content?(text, wait: 3)
 
-  repeat_until_timeout(message: "Couldn't find text '#{text}'") do
-    break if has_content?(text, wait: 3)
+  begin
+    repeat_until_timeout(message: "Couldn't find text '#{text}'") do
+      break if has_content?(text, wait: 3)
 
-    begin
-      accept_prompt do
-        execute_script 'window.location.reload()'
-      end
-    rescue Capybara::ModalNotFound
-      # ignored
+      refresh_page
     end
+  rescue ScriptError, TimeoutError => e
+    # an error may be raised while the page still (re)loads, making no screenshot available
+    find('#page-body', wait: 3)
```

Comment:
Maybe instead of waiting here, we should be waiting in the After method, just before we take the screenshot.

Reply:
Right, it's much better indeed to have this kind of stuff in the After method.

Regarding the waits: when  the failure happens in cases were we are not dealing with bootstrapping minions (as when checking the events history), this and condition will probably short circuit due to lazy evaluation.

`click_button('Details') if has_content?('Bootstrap Minions') && has_content?('Details')`

Maybe we can check for the exception which lead to a failure and use it to determine whether to force another wait, or not.
Something like:

`find('#page-body', wait: 3) if scenario.exception.is_a?(TimeoutError) `

Assuming the problem is actually not having enough time for the page to load.